Set up analytics panel and preference persistence

etanshaul opened the pull request:
more progress towards #47 

- Creates the Google menu item and the Usage Tracker menu item in the IDE settings\
- Persists the user preference

![image](https://user-images.githubusercontent.com/1735744/50795154-4ea24900-129b-11e9-8abe-f7b9d0c6baaf.png)

**Interop with Google Cloud Tools plugin**
The GCT plugin also defines these menu items. As long as the IDs of the configured panels match, the platform will ensure that _only a single instance of the menu item will be created_. The IDs are set as follows:
"Google" menu item -> "google.settings"
"Usage Tracking" menu item -> "google.usagetracker.settings"

These IDs must be set identically in both plugins. With this, when both plugins are installed, only one of the plugins will contribute the menu items (based on the loading order of the plugins). Since both plugins persist the user preference with the same key, the selection is effectively shared across the plugins.

TODO:
- Update the GCT plugin's usage tracker menu item to have the same ID (the "Google" menu item already has an ID): https://github.com/GoogleCloudPlatform/google-cloud-intellij/issues/2316
- Setup the usage tracking notification infrastructure (on first install of the plugin) - next PR

etanshaul commented:
@patflynn to look at the wording. I will update the wording on the GCT plugin to match the wording we finalize here.

ivanporty commented on core/src/main/kotlin/com/google/container/tools/core/util/UsageTrackerBundle.kt:
```diff
@@ -0,0 +1,38 @@
+/**
+ * Message bundle manager for usage tracking related featured.
+ */
+object UsageTrackerBundle {
```
Could we possibly merge the two bundles (core and usage tracker) into one bundle, probably just `core`? It feels very fine grained and mildly confusing since the core has only one string and probably the usage tracker will stay in the core module anyway? I believe we usually chose to have a single bundle per module in our cloud tools plugin.

ivanporty commented on core/src/main/kotlin/com/google/container/tools/core/settings/UsageTrackerConfigurable.kt:
```diff
@@ -0,0 +1,98 @@
+            Insets(0, 0, 0, 0)
+        )
+
+        return panel {
```
Could we return `usageTrackingPanel` directly instead of additional panel wrapper?

etanshaul replied:
I tried, but I couldn't get the titled panel thing to appear (where is says "Usage Tracking"). I suspect this is because the outer panel is the entire panel of the menu item (and no title is allowed)? Not 100% sure why.

ivanporty commented on core/src/main/kotlin/com/google/container/tools/core/analytics/UsageTrackerManagerService.kt:
```diff
@@ -18,32 +18,42 @@ package com.google.container.tools.core.analytics
  */
-class UsageTrackerManagerService {
+class UsageTrackerManagerService(
+    private val trackingPreferenceProperty: PropertiesComponent = PropertiesComponent.getInstance()
```
It might be useful to add a short parameter style comment for this, i.e. how is this used in a few words.

etanshaul replied:
done

ivanporty commented on core/src/test/kotlin/com/google/container/tools/core/settings/UsageTrackerConfigurableTest.kt:
```diff
@@ -0,0 +1,88 @@
+        usageTrackerConfigurable = UsageTrackerConfigurable()
+    }
+
+    @Test
```
since UI state is modified here, it makes sense to use `@UITest` as well to protect against possible racing

ivanporty commented on core/src/test/kotlin/com/google/container/tools/core/settings/UsageTrackerConfigurableTest.kt:
```diff
@@ -0,0 +1,88 @@
+        Truth.assertThat(usageTrackerConfigurable.isModified).isTrue()
+    }
+
+    @Test
```
ditto `@UITest`

ivanporty commented on core/src/test/kotlin/com/google/container/tools/core/settings/UsageTrackerConfigurableTest.kt:
```diff
@@ -0,0 +1,88 @@
+        Truth.assertThat(usageTrackerConfigurable.isModified).isFalse()
+    }
+
+    @Test
```
ditto @UITest

ivanporty commented on core/src/test/kotlin/com/google/container/tools/core/settings/UsageTrackerConfigurableTest.kt:
```diff
@@ -0,0 +1,88 @@
+        verify { usageTrackerManagerService.setTrackingOptedIn(true) }
+    }
+
+    @Test
```
ditto `@UITest`

etanshaul commented:
@ivanporty thanks for the review. I think I addressed your comments so far.

Still need to get a solution to [my comment](https://github.com/GoogleContainerTools/google-container-tools-intellij/pull/119#discussion_r246067553)

etanshaul commented:
I'll address [that issue ](https://github.com/GoogleContainerTools/google-container-tools-intellij/pull/119#issuecomment-452362034) in the next PR so that we can focus more clearly on it
